### DISCUSSION TITLE
Modifying federated learning

### DISCUSSION THREAD

**Discussion by Nicholas-B1** [NVIDIA/NVFlare#1900] [Q&A]
Hello, I would like to know how I would go about cutting down datasets or features or CNN layers to make faster running federated learning. I want to be able to test the speed of these examples like the cifar10 example but make it run faster even if it means sacrificing some accuracy.

**Comment by YuanTingHsieh** [comment]
This is a general research question, @holgerroth @ZiyueXu77 can you help sharing your thoughts on this discussion, thanks!

**Comment by holgerroth** [comment]
Hi @Nicholas-B1 , you can reduce the number of layers or use the [SimpleCNN](https://github.com/NVIDIA/NVFlare/blob/af74044c643f1fd2935d3867c9a77e0be4ab2cd6/examples/advanced/cifar10/pt/networks/cifar10_nets.py#L45) in the network definitions. Also, changing the number of training samples being loaded in the data loader should speed things up. The easiest would be to break the loop after a certain number of iterations in [training](https://github.com/NVIDIA/NVFlare/blob/af74044c643f1fd2935d3867c9a77e0be4ab2cd6/examples/advanced/cifar10/pt/learners/cifar10_model_learner.py#L220) and [validation](https://github.com/NVIDIA/NVFlare/blob/af74044c643f1fd2935d3867c9a77e0be4ab2cd6/examples/advanced/cifar10/pt/learners/cifar10_model_learner.py#L341).
